src/profiles_management/pmr/classes.py

```diff
@@ -0,0 +1,195 @@
+"""This package provides classes for validating and defining a PMR.
+
+The goal of these classes is to represent the PMR in a more
+Pythonic way, rather than a dict of vavlues.
+
+The classes should also provide some common helper methods for
+traversing the information of the PMR, but not functions for
+directly affecting the K8s cluster based on the PMR.
+
+This is similar to how the K8s python clients have classes
+for representing the objects and different methods for doing operations
+with them.
+"""
+
+import logging
+from enum import StrEnum
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, TypeAdapter
+
+log = logging.getLogger(__name__)
+
+
+# Classes for ResourceQuotaSpec
+class Operator(StrEnum):
+    """Pydantic class for the Operator in MetchExpression of ResourceQuotaSpec."""
+
+    In = "In"
+    NotIn = "NotIn"
+    Exists = "Exists"
+    DoesNotExist = "DoesNotExist"
+
+
+class ScopedResourceSelectorRequirement(BaseModel, extra="forbid"):
+    """Pydantic class for objects of matchExpressions of ResourceQuotaSpec.
+
```

Comment:
We must also explain what each of the attributes are, for example

```
  Args:
    operator (Operator): it is ? 
    scope_name (str): it is ?
```

This comment applies in other classes as well.

Reply:
eb10177

Note that I didn't describe the types, as all attributes already contain types and LSP servers can show this info in editors.